src/libraries/System.Runtime.Numerics/src/System/Numerics/IeeeDecimalNumber.Conversions.cs

```diff
@@ -0,0 +1,5085 @@
+            // which in this case is e >= ceil(128 * log_10(2)) = 39
+
+            if (e >= 39)
+            {
+                // PORT: Removed support for status flag setting
+                return return_binary32_inf(s);
+            }
+            // Also check for "trivial" underflow, when 10^e * 2^113 <= 2^emin * 1/4,
+            // so test e <= floor((emin - 115) * log_10(2))
+            // In this case just fix ourselves at that value for uniformity.
+            //
+            // This is important not only to keep the tables small but to maintain the
+            // testing of the round/sticky words as a correct rounding method
+
+            if (e <= -80)
+            {
+                e = -80;
+            }
+
+            // Look up the breakpoint and approximate exponent
+
+            m_min = (bid_breakpoints_binary32[2 * (e + 80)], bid_breakpoints_binary32[2 * (e + 80) + 1]);
+            e_out = bid_exponents_binary32[e + 80] - k;
+
+            // Choose provisional exponent and reciprocal multiplier based on breakpoint
+            if (c.Item2 <= m_min.Item2)
+            {
+                r = (bid_multipliers1_binary32[4 * (e + 80)],
+                    bid_multipliers1_binary32[4 * (e + 80) + 1],
+                    bid_multipliers1_binary32[4 * (e + 80) + 2],
+                    bid_multipliers1_binary32[4 * (e + 80) + 3]);
+            }
+            else
+            {
+                r = (bid_multipliers2_binary32[4 * (e + 80)],
+                    bid_multipliers2_binary32[4 * (e + 80) + 1],
+                    bid_multipliers2_binary32[4 * (e + 80) + 2],
+                    bid_multipliers2_binary32[4 * (e + 80) + 3]);
+                e_out++;
+            }
+
+            // Do the reciprocal multiplication
+            __mul_64x256_to_320(out z, c.Item2, r);
+            z.Item6 = z.Item5; z.Item5 = z.Item4; z.Item4 = z.Item3; z.Item3 = z.Item2; z.Item2 = z.Item1; z.Item1 = 0;
+
+            // Check for exponent underflow and compensate by shifting the product
+            // Cut off the process at precision+2, since we can't really shift further
+            if (e_out < 1)
+            {
+                int d;
+                d = 1 - e_out;
+                if (d > 26)
+                    d = 26;
+                e_out = 1;
+                srl256_short(ref z.Item6, ref z.Item5, ref z.Item4, ref z.Item3, d);
+            }
+            c_prov = z.Item6;
+
+            // Round using round-sticky words
+            // If we spill into the next binade, correct
+            // Flag underflow where it may be needed even for |result| = SNN
+
+            if (lt128(
+                bid_roundbound_128[((s & 1) << 1) + (int)(c_prov & 1)].Item2, // PORT: the calculation of this index factors in the rounding mode in the original code
+                bid_roundbound_128[((s & 1) << 1) + (int)(c_prov & 1)].Item1,
+                z.Item5,
+                z.Item4
```

Comment:
I will also generally note that porting this code can be *very* error prone. I happened to discover a typo I made last week in which this line was incorrectly `z.Item3` instead of `z.Item4`. If I hadn't found it by chance, I am not sure how we would have ever found that bug. Even with exhaustive tests (which are still up in the air) I'd imagine debugging this code is a nightmare. We should keep this in mind when thinking about development cost.